pkg/pillar/zedcloud/send.go

```diff
@@ -350,20 +417,20 @@ func SendOnIntf(workContext context.Context, ctx *ZedCloudContext, destURL strin
 			log.Traceln(errStr)
 			return nil, nil, senderStatus, errors.New(errStr)
 		}
-		errStr := fmt.Sprintf("No IP addresses to connect to %s using intf %s",
-			reqUrl, intf)
-		log.Traceln(errStr)
-		return nil, nil, senderStatus, errors.New(errStr)
+		err = &types.IPAddrNotAvail{IfName: intf}
+		log.Trace(err)
+		return nil, nil, senderStatus, err
 	}
 	dnsServers := types.GetDNSServers(*ctx.DeviceNetworkStatus, intf)
 	if len(dnsServers) == 0 {
 		if ctx.FailureFunc != nil {
 			ctx.FailureFunc(log, intf, reqUrl, 0, 0, false)
 		}
-		errStr := fmt.Sprintf("No DNS servers to connect to %s using intf %s",
-			reqUrl, intf)
-		log.Traceln(errStr)
-		return nil, nil, senderStatus, errors.New(errStr)
+		err = &types.DNSNotAvail{
+			IfName: intf,
+		}
+		log.Trace(err)
```

Comment:
This will call the String() method on DNSNotAvail which does not include the IfName, 
Don't know if it makes sense to have all of those String() methods you defined include the IfName but here we are definitely loosing information in the log.Trace.

Reply:
Added interface name to the error message of DNSNotAvail

Reply:
Yes